common/djangoapps/third_party_auth/decorators.py

```diff
@@ -30,3 +34,40 @@ def wrapped_view(request, *args, **kwargs):
         resp['X-Frame-Options'] = x_frame_option
         return resp
     return wraps(view_func, assigned=available_attrs(view_func))(wrapped_view)
+
+
+from urllib import urlencode
+def tpa_hint_ends_existing_session(func):
+
+    @wraps(func)
+    def inner(request, *args, **kwargs):
+        sso_provider = None
+        provider_id = request.GET.get('tpa_hint')
+        logged_out = request.GET.get('logged_out') == 'yes'
```

Comment:
It gets added to the redirect URL by this decorator - basically, it's a way to detect "I've already done my stuff to this request sequence".